kpops/components/streams_bootstrap/producer/producer_app.py

```diff
@@ -19,7 +21,10 @@ class ProducerApp(KafkaApp):
     This producer holds configuration to use as values for the streams bootstrap produce helm chart.
     """
 
-    _type = "producer"
+    type: ClassVar[str] = "producer"
+    schema_type: Literal["producer"] = Field(
```

Comment:
Just for curiosity, What is the difference between `Literal` and `Final` in python? Could you define schema_type `Final`?

Reply:
`Literal` is used for the discriminator (differentiation of models) here as it is a literal string value. It's needed for the schema generation, so that it can select the right model schema based on the `type` of a pipeline step. 
`Final` I haven't used so far, but from my understanding it is used as a special indicator for type checkers to tell that the attribute can never be overridden.